cmd/internal/ecs/add.go

```diff
@@ -0,0 +1,330 @@
+		return nil
+	} else if _, ok := err.(UsageError); ok {
+		telemetry.Error("Add to ECS", err)
+		return err
+	} else {
+		telemetry.Error("Add to ECS", err)
+		return cmderr.AkitaErr{Err: err}
+	}
+	return err
+}
+
+// Initial state: check if running interactively, if so then start
+// with collecting AWS profile.
+func initState(wf *AddWorkflow) (keepGoing bool, err error) {
+	reportStep("Start Add to ECS")
+
+	// Check if running interactively.
+	// TODO: I didn't see a way to do this from go-survey directly.
+	if !term.IsTerminal(int(os.Stdin.Fd())) {
+		return fillFromFlags(wf)
+	}
+
+	wf.currentState = getProfileState
+	return true, nil
+}
+
+// Ask the user to specify a profile; "" is fine to use the default profile.
+// TODO: it seems very difficult to present a list (which is what I was trying
+// to do orginally) because the SDK doesn't provide an API to do that, and
+// its config file parser is internal.
+func getProfileState(wf *AddWorkflow) (keepGoing bool, err error) {
+	reportStep("Get AWS Profile")
+
+	if awsProfileFlag != "" {
+		wf.awsProfile = awsProfileFlag
+		if err = wf.checkCredentials(); err != nil {
+			printer.Errorf("Could not find AWS credentials for profile %q. The error from the AWS library is shown below.\n")
+			return false, errors.Wrap(err, "Error loading credentials")
+		}
+
+		wf.currentState = getRegionState
+		return true, nil
+	}
+
+	// Use the existing value as the default in case we repeat this step
+	err = survey.AskOne(
+		&survey.Input{
+			Message: "Which of your AWS profiles should Akita use to configure ECS?",
+			Help:    "Enter the name of the AWS profile you use for configuring ECS, or leave blank to try the default profile. Akita needs this information to identify which AWS credentials to use.",
+			Default: wf.awsProfile,
+		},
+		&wf.awsProfile,
+	)
+	if err != nil {
+		return false, err
+	}
+
+	wf.awsCred = credentials.NewSharedCredentials(awsCredentialsFlag, wf.awsProfile)
+	if err = wf.checkCredentials(); err != nil {
+		printer.Errorf("Error from AWS library: %v\n", err)
+		printer.Errorf("Could not find AWS credentials for profile %q, please try again or hit Ctrl+C to exit.\n", wf.awsProfile)
+		wf.awsProfile = ""
+		return true, nil
+	}
+
+	// TODO: I don't think there's even a way to show the profile name here?
+	printer.Infof("Successfully loaded AWS credentials.\n")
+
+	wf.currentState = getRegionState
+	return true, nil
+}
+
+const findAllClusters = "find all clusters"
+
+var publicAWSRegions = []string{
+	"default",
+	findAllClusters,
+	"af-south-1",
+	"ap-east-1",
+	"ap-northeast-1",
+	"ap-northeast-2",
+	"ap-northeast-3",
+	"ap-south-1",
+	"ap-southeast-1",
+	"ap-southeast-2",
+	"ap-southeest-3",
+	"ca-central-1",
+	"eu-central-1",
+	"eu-central-1",
+	"eu-north-1",
+	"eu-south-1",
+	"eu-west-1",
+	"eu-west-2",
+	"eu-west-3",
+	"me-central-1",
+	"me-south-1",
+	"sa-east-1",
+	"us-east-1",
+	"us-east-2",
+	"us-west-1",
+	"us-west-2",
+}
```

Comment:
Surely, we can get this list programmatically, so that we don't have to maintain it.

Reply:
That's what I said!  But I haven't found it yet. Maybe it's in one of the services.

I'll take another look or do some google searches.

Reply:
This looks promising: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2#Client.DescribeRegions

Reply:
I left the list as a fallback, because it is entirely possible not to have DescribeRegions permissions.